### ISSUE TITLE
Info bubble with "invalid storage backend settings" when save settings in admin panel

### ISSUE THREAD

**Issue by nupplaphil** [friendica/friendica#9402]
I get an info bubble with "invalid storage backend settings" now when I save the settings in admin panel.

_Originally posted by @AlfredSK in https://github.com/friendica/friendica/pull/9373#issuecomment-705841947_

**Comment by nupplaphil** [comment]
@AlfredSK I cannot reproduce it when saving the "settings".

For `storage.name`:
- What're your config settings in a file (like `local.config.php`) ?
- What're your config settings in the DB?
- What's the output for these settings per `console.php config storage name` ?

What's your config adapter ? (`system.config_adapter`)

**Comment by AlfredSK** [comment]
Do you need *everything* from the config table? It has 510 rows. :-)

Config adapter is PRELOAD.


> What's the output for these settings per console.php config storage name ?
```
php bin/console.php config storage name
storage.name => Filesystem
```

**Comment by AlfredSK** [comment]
First, a screenshot:

![storage-settings](https://user-images.githubusercontent.com/3656054/95626639-f6553800-0a7a-11eb-8e52-00d4d3f3596a.png)

**Comment by AlfredSK** [comment]
Config file w/o the database Gedöhns...

```
        'config' => [
                'admin_email' => 'xxxxxxxx@libranet.de, xxxxxxxx@libranet.de',
                'admin_name' => 'Steffen K9 🐰',
                'sitename' => 'LIBRANET.de',
                'register_policy' => \Friendica\Module\Register::APPROVE,
                'register_text' => '',
                'php_path' => '/usr/bin/php',
                'max_import_size' => 256000,
        ],
        'system' => [
                'default_timezone' => 'UTC',
                'language' => 'en',
                'basepath' => '/var/www/html/libranet',
                'local_search' => false,
                'update_active_contacts' => true,
                'proxy_cache_time' => 432000,
                'min_poll_intervall' => 20,
                'max_feed_items' => 20,
                'throttle_limit_day' => 40,
                'curl_range_bytes' => 15728640,
                'worker_debug' => false,
                'profiler' => false,
                'worker_jpm' => true,
                'worker_fetch_limit' => 4,
                'cache_driver' => 'redis',
                'redis_host' => '/var/run/redis/redis.sock',
                'redis_port' => 0,
                'redis_db' => 2,
                'session_handler' => 'cache',
                'ignore_cache' => false,
                'like_no_comment' => false,
                'big_emojis' => true,
                'png_quality' => 6,
                'auth_cookie_lifetime' => 7,
                'dbclean-expire-limit' => 350,
                'config_adapter' => 'preload',
        ],
        'theme' => [
                'hide_eventlist' => true,
        ],
```

**Comment by AlfredSK** [comment]
Database from table config:

```
MariaDB [friendica]> select cat, k, v from config where cat like 'storage';
+---------+-----------------+-----------------------+
| cat     | k               | v                     |
+---------+-----------------+-----------------------+
| storage | filesystem_path | /var/www/libranetdata |
| storage | name            | Filesystem            |
+---------+-----------------+-----------------------+
2 rows in set (0.000 sec)
```

**Comment by AlfredSK** [comment]
Btw. the same error notification exists on both my develop nodes.

**Comment by AlfredSK** [comment]
And these are the logged messages I get *every time* when I save the settings at /admin/site:

```
2020-10-09T20:27:50Z index [INFO]: DELETE FROM `config`  WHERE (`cat` = 'system' AND `k` = 'email_banner') {"callstack":"Config::delete, PreloadConfig::delete, Site::post, call_user_func, Module::run, App::runFrontend","internal_callstack":{"config:{\"cat\":\"system\",\"k\":\"email_banner\"}":true}} - {"file":"Database.php","line":1311,"function":"delete","uid":"0505ce","process_id":2671298}
2020-10-09T20:27:50Z index [INFO]: installed theme frio [] - {"file":"theme.php","line":61,"function":"frio_install","uid":"0505ce","process_id":2671298}
2020-10-09T20:27:50Z index [INFO]: DELETE FROM `config`  WHERE (`cat` = 'system' AND `k` = 'mobile-theme') {"callstack":"Config::delete, PreloadConfig::delete, Site::post, call_user_func, Module::run, App::runFrontend","internal_callstack":{"config:{\"cat\":\"system\",\"k\":\"mobile-theme\"}":true}} - {"file":"Database.php","line":1311,"function":"delete","uid":"0505ce","process_id":2671298}
2020-10-09T20:27:50Z index [INFO]: DELETE FROM `config`  WHERE (`cat` = 'system' AND `k` = 'singleuser') {"callstack":"Config::delete, PreloadConfig::delete, Site::post, call_user_func, Module::run, App::runFrontend","internal_callstack":{"config:{\"cat\":\"system\",\"k\":\"singleuser\"}":true}} - {"file":"Database.php","line":1311,"function":"delete","uid":"0505ce","process_id":2671298}
```

**Comment by nupplaphil** [comment]
> Do you need _everything_ from the config table? It has 510 rows. :-)
> 
> Config adapter is PRELOAD.
> 
> > What's the output for these settings per console.php config storage name ?
> 
> ```
> php bin/console.php config storage name
> storage.name => Filesystem
> ```

Oh, no I just needed the `storage.name` config entry .. I think I wasn't precise enough with my intro "for `storage.name`:" ... Thx for the input .. wow :D I think that's enough ;-)

**Comment by nupplaphil** [comment]
I've exactly the same storage config like you and still .. there's no bubble :-/

But at least I can see the config delete commands like at https://github.com/friendica/friendica/issues/9402#issuecomment-706389112 as well ..

**Comment by annando** [comment]
This message only appears with old systems when the storage is still set to the old storage method directly in the `photo` table. It disappears when another storage method is selected.

**Comment by AlfredSK** [comment]
What do you mean with "another storage method". There is no other method to select. 'Database' or 'Filesystem'. My setting is 'Filesystem'.

**Comment by annando** [comment]
I had this message only when I hadn't either selected "Database" nor "Filesystem" but storing in the `photo` table.

**Comment by AlfredSK** [comment]
Aha. As you can see in the screenshot above and in the database query result all settings are correct (as far as I can tell). I have the same issue on all 4 nodes I am running.

Also, it has been confirmed by other admins in the forum.

**Comment by MrPetovan** [comment]
Hey, my system isn't old, it's "experienced".

**Comment by AlfredSK** [comment]
I tried to switch it back and forth between Database and Filesystem but the notification bubble with the error message still appears.

**Comment by annando** [comment]
Okay. So your system is to blame. :grin: 

I will have a look at it in my current (issue) clean up job.

**Comment by AlfredSK** [comment]
Btw... the error message also appears when I save the 'Database' storage setting. How can this be invalid? I'm puzzled.

**Comment by AlfredSK** [comment]
Sorry for the bad news but it is not fixed. Please reopen.

The message still appears when hitting any save/submit button at `.../admin/site`.

**Comment by pztrn** [comment]
Confirming, #9680 fixes empty selection for storage.

But I can't change to another with the very same message - "invalid storage backend settings". Trying to change from database to filesystem.

**Comment by rnwgnr** [comment]
I get this message too every time i update the settings, but only if i use the `preload` config adapter. As soon as i switch to `jit` the message is gone.

**Comment by annando** [comment]
@guzzisti So it is some ugly side effect. Thanks for the information.

**Comment by MrPetovan** [comment]
Closed by #10104
